py_neuromodulation/features/coherence.py

```diff
@@ -83,10 +83,14 @@ def get_coh(self, feature_results, x, y):
         self.Pyy = welch(y, self.sfreq, self.window, nperseg=128)[1]
         self.Pxy = csd(x, y, self.sfreq, self.window, nperseg=128)[1]
 
-        if self.coh:
-            self.coh_val = np.abs(self.Pxy**2) / (self.Pxx * self.Pyy)
-        if self.icoh:
-            self.icoh_val = np.array(self.Pxy / (self.Pxx * self.Pyy)).imag
+        if self.coh and self.icoh:
+            cohy = self.Pxy / np.sqrt(self.Pxx * self.Pyy)
+            self.coh_val = np.abs(cohy)
+            self.icoh_val = cohy.imag
+        elif self.coh:
+            self.coh_val = np.abs(self.Pxy ** 2) / (self.Pxx * self.Pyy)
+        else:  # only self.icoh == True
+            self.icoh_val = self.Pxy.imag / np.sqrt(self.Pxx * self.Pyy)
```

Comment:
Just a suggestion, but if both coh and icoh are requested features, then I expect it is faster to compute coherency once, then extract coh and icoh values. Otherwise if only one is requested, it can follow the same path as before of computing this information separately.